### PR TITLE
fix: batch write

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

**Review by rolsonquadras — APPROVED** [review]

**Review by soluchok — APPROVED** [review]

**Review by DRK3 — APPROVED** [review]

File: pkg/storage/formattedstore/batchwrite.go [hunk 1]
```diff
@@ -176,35 +179,30 @@ func (p *pendingBatch) put(addBatch *batch) error {
...
 
-	p.mutex.Unlock()
-
 	return nil
 }
 
 func (p *pendingBatch) delete(addBatch *batch) error {
 	p.size++
 
+	p.mutex.Lock()
```
Comment (troyronda) [inline pkg/storage/formattedstore/batchwrite.go:193]: Interesting - GenerateEDVDocumentID needs a lock around it?

**Review by troyronda — APPROVED** [review]

File: pkg/storage/formattedstore/batchwrite.go [hunk 2]
```diff
@@ -176,35 +179,30 @@ func (p *pendingBatch) put(addBatch *batch) error {
...
 	return nil
 }
 
 func (p *pendingBatch) delete(addBatch *batch) error {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
 	p.size++
 
 	id, err := p.formatter.GenerateEDVDocumentID(addBatch.keyID)
```
Comment (troyronda) [inline pkg/storage/formattedstore/batchwrite.go:196]: Note: might be worth commenting on why p.formatter.GenerateEDVDocumentID depends on state.

**Review by DRK3 — COMMENTED** [review]
Do you still have multiple threads? If not... then do you need the locks at all?

**Comment by fqutishat** [comment]
> Do you still have multiple threads? If not... then do you need the locks at all?

for flush

**Pull request merged**